fix settings avform

antis81 opened the pull request:
Fix some things in AVForm:
- [x] initialize audio output volume from settings (fixes #3478)
- [x] rename ui file and class -> it integrates better with the code, e.g. allows switch between ui/code etc.
- [x] fix type conversions and comparisons concerning video resolution and frame rate

zetok commented:
> initialize audio output volume from settings -> requires some additional work (fixes #3480)

Not #3478 ?

antis81 commented:
Erm, yep. Referenced the wrong PR by accident. It is going to fix #3478.Corrected in description!

antis81 committed:
refactor(avform): improve code flow

* rename file avsettings.ui -> avform.ui, introducing seamless switching between UI/Code
* switch to "private multi-inheritance" pattern, which has several advantages

antis81 committed:
refactor(avform): fixed type conversion warnings and coding style

antis81 committed:
fix(avform): initialize slider value from settings

antis81 committed:
fix(settings): block signals during initialization

Prevents unexpected writes to settings during object creation, which can potentially mess up various settings values.

sudden6 commented on src/widget/form/settings/avform.h:
```diff
@@ -61,7 +60,7 @@ private slots:
     void onAudioInDevChanged(int deviceIndex);
     void onAudioOutDevChanged(int deviceIndex);
-    void onPlaybackValueChanged(int value);
+    void on_playbackSlider_valueChanged(int value);
```
this line doesn't quite match our naming conventions.

antis81 replied:
It's an autoconnection, it has to be this way. Others will change the same.

antis81 committed:
refactor(avform): use auto-connections in ui signals

antis81 commented:
@sudden6 Ready for merge.

zetok commented:
on 4e56494d46ac2e72e9b072924c5196c0d9fcb2c2

No idea whether related to the PR; buttons are green and don't indicate that there's an ongoing call, not sure what caused this, I've went into settings & rescanned devices & set newly connected audio device as input.
![buttons](https://cloud.githubusercontent.com/assets/3148759/16873130/29f60a1a-4a89-11e6-87a3-39960c11e404.png)

Note that I can't toggle muting microphone / speakers when buttons don't indicate that there's an ongoing call.

Also, after switching to the new audio device there's no longer any voice in the call – voice is (transmitted?) only when I have A/V settings tab open, otherwise there's only silence; according to `pavucontrol` qTox doesn't even record audio during the call(!).

antis81 commented:
> No idea whether related to the PR; buttons are green and don't indicate that there's an ongoing call, not sure what caused this, I've went into settings & rescanned devices & set newly connected audio device as input.

The observed behaviour is completely unrelated to this PR.

zetok commented:
> The observed behaviour is completely unrelated to this PR.

yep, #3521
